content/blog/2018-04-03-rtika-introduction.md

```diff
@@ -0,0 +1,277 @@
+
+#### Motivation: the Dreaded `.lob` File Extension
+
+This package came together when parsing Word documents in a governmental archive. The files did not have a helpful file extension. They had been stored as 'large object data' in a database, and given the generic `.lob` extension. Some documents parsed with the `antiword` package:
+
+``` r
+library('antiword')
+
+timing <- system.time(
+  
+  text <- 
+    batch[1:2000] %>%
+    lapply(antiword)
+  
+)
+
+# average time elapsed *per document* for antiword:
+timing[3]/2000
+#>   elapsed 
+#> 0.0098275
+```
+
+However, the files farther into the batch were in the new Word format, and `antiword` did not parse them. The government had switched to the new version but left no obvious clues. Typically, modern Word documents use the `.docx` file extension and the ancient ones use `.doc`. I just had `.lob`:
+
+``` r
+last_file <- length(batch)
+tryCatch(antiword(batch[last_file]), error = function(x){x})
+#> <simpleError: System call to 'antiword' failed (1): /Users/sasha/leginfo/2017/BILL_ANALYSIS_TBL_9999.lob is not a Word Document.
+```
+
+Fortunately, I remembered Apache Tika. Five years earlier, Tika helped parse the Internet Archive, and handled whatever format I threw at it. Back then, I put together a R package for myself and a few colleagues. It was outdated.
+
+I downloaded Tika and made a R script. Tika did its magic. It scanned the headers for the "Magic Bytes" [^4] and parsed the files appropriately:
+
+``` r
+library('rtika')
+
+timing <- system.time(
+  
+  text <- 
+    batch[1:2000] %>%
+    tika_text(threads=1)
+  
+)
+
+# average time elapsed *per document* parsed for rtika:
+timing[3]/2000
+#>  elapsed 
+#> 0.006245
+```
+
+For this batch, the efficiency compared favorably to `antiword`, even with the overhead of loading Tika. I estimate that starting Tika, loading the Java parsers each time, loading the file list from R, and reading the files back into an R object took a few extra seconds. The reduced time effort processing the entire batch led me to think about the broader applications of Tika. This was too good not to share, but I was apprehensive about maintaining a package over many years. The rOpenSci organization was ready to help.
+
+#### Lessons Learned
+
+I never distributed a package before on repositories such as CRAN or Github, and the rOpenSci group was the right place to learn how. The reviewers used a transparent on-boarding process and taught about good documentation and coding style. They were helping create a maintainable package by following certain standards. If I stopped maintaining `rtika`, others could use their knowledge of the same standards to take over. The vast majority of time was spent on documenting the code, the introductory vignette, and continuous testing to integrate new code.
```

Comment:
onboarding, no hyphen

Reply:
consider linking to http://onboarding.ropensci.org/ or https://github.com/ropensci/onboarding so others can see what it's about (I've asked in #onboarding channel which is link is preferable from blog post)

Reply:
In "The reviewers used a transparent on-boarding process and taught about good documentation and coding style." you might link to the open review thread: https://github.com/ropensci/onboarding/issues/191

Reply:
confirmed: please link to https://github.com/ropensci/onboarding